tests/Common/BranchMetadataTest.php

```diff
@@ -140,6 +153,110 @@ public function testSetMetadataEvent()
+        /** @var array $deletedMetadata */
+        $deletedMetadata = $defaultMdClient->listBranchMetadata();
+        $this->assertCount(1, $deletedMetadata);
+        // check there is not deleted one
+        $this->assertNotSame($metadata[0]['id'], $deletedMetadata[0]['id']);
+        $this->assertMetadataEquals(self::TEST_METADATA[1], $deletedMetadata[0]);
+
+        // delete metadata - first again
+        try {
+            $defaultMdClient->deleteBranchMetadata($metadata[0]['id']);
+            $this->fail('should fail');
+        } catch (ClientException $e) {
+            $this->assertContains(sprintf('Metadata with id "%s" not found', $metadata[0]['id']), $e->getMessage());
+            $this->assertSame(404, $e->getCode());
+        }
+    }
+
+    /**
+     * @return void
+     */
+    public function testDeleteMetadataRestrictionForReadOnlyUser()
+    {
+        // create read only client
+        $readOnlyClient = $this->clientProvider->getDefaultBranchClient([
+            'token' => STORAGE_API_READ_ONLY_TOKEN,
+            'url' => STORAGE_API_URL,
+        ]);
+        // create metadata client
+        $readOnlyMdClient = new DevBranchesMetadata($readOnlyClient);
+        $defaultMdClient = new DevBranchesMetadata($this->client);
+
+        // list metadata
+        /** @var array $metadata */
+        $metadata = $readOnlyMdClient->listBranchMetadata();
+        $this->assertCount(0, $metadata);
+
+        // add metadata to delete
+        /** @var array $metadata */
+        $metadata = $defaultMdClient->postBranchMetadata(self::TEST_METADATA);
+        $this->assertCount(2, $metadata);
+
+        // delete metadata
+        try {
+            $readOnlyMdClient->deleteBranchMetadata($metadata[0]['id']);
+            $this->fail('should fail, insufficiently permission');
+        } catch (ClientException $e) {
+            $this->assertContains("You don't have access to resource", $e->getMessage());
+            $this->assertSame(403, $e->getCode());
+        }
+    }
+
+    /**
+     * @return void
+     */
+    public function testDeleteMetadataEvent()
+    {
+        // create metadata client
+        $defaultMdClient = new DevBranchesMetadata($this->client);
+
+        // add metadata
+        /** @var array $metadata */
+        $metadata = $defaultMdClient->postBranchMetadata(self::TEST_METADATA);
+        $this->assertCount(2, $metadata);
+
+        // delete metadata - first
+        $defaultMdClient->deleteBranchMetadata($metadata[0]['id']);
+
+        $events = $this->listEvents($this->client, 'storage.devBranchMetadataDeleted');
+
+        $this->assertEvent(
+            $events[0],
+            'storage.devBranchMetadataDeleted',
+            sprintf('Development branch "%s" metadata with key "%s" deleted', 'Main', 'KBC.SomeEnity.metadataKey'),
+            $this->client->getCurrentBranchId(),
+            'Main',
+            'devBranch',
+            [
+                'metadata' => self::TEST_METADATA,
```

Comment:
Pustil sis ten test lokálně vůbec? Tady to asi těžko može někdy projít. Když v eventu kam dáváš `metadataId` `key` čekáš původní metadata.
![image](https://user-images.githubusercontent.com/180148/154640431-04c57128-29ef-402b-a38f-a837abf0f101.png)

Reply:
fixnuto